xlators/features/bit-rot/src/bitd/bit-rot-scrub.c

```diff
@@ -1318,6 +1318,12 @@ br_scrubber_scale_up(xlator_t *this, struct br_scrubber *fsscrub,
     if ((i != diff) && !scrub)
         goto error_return;
 
+    if (ret && i == 0 && scrub) {
```

Comment:
i < diff ?

It can still fail scheduling scrub after scheduling first one!

Reply:
@kotreshhr I have one question, in every iteration of the loop at line 1034 "diff" number of scrubbers are being allocated. Is that intentional? I think we should allocate only "1" in every iteration.

Reply:
Implemented the changes of freeing scrub before break and allocating only a single scrubber instead of diff number of scrubbers since it is inside the loop.

For the time being I have kept the way it returns the error since it should not have any negative impact on scrubbing if it fails to throttle up to the desired number of scrubbers. Also the behavior is same in case of scaling down as well. So in both the places if it succeeds to scale{up/down} at least one scrubber it returns success with a warning message.